contracts/pcv/utils/FeiSink.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "../IPCVDeposit.sol"; 
+import "../../refs/CoreRef.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+/// @title a contract to skim excess FEI from addresses
+/// @author Fei Protocol
+contract FeiSink is CoreRef {
+    using EnumerableSet for EnumerableSet.AddressSet;
+ 
+    event ThresholdUpdate(address source, uint256 newThreshold);   
+    event SourceAdded(address newSource, uint256 threshold);
+    event SourceRemoved(address source);
+
+    /// @notice source PCV deposit to skim excess FEI from
+    EnumerableSet.AddressSet private sources;
+
+    /// @notice the threshold of FEI above which to skim
+    mapping(address => uint256) public thresholds;
+
+    /// @notice FEI Skimmer
+    /// @param _core Fei Core for reference
+    /// @param _sources the targets to skim from
+    /// @param _thresholds the thresholds of FEI to be maintained by sources
+    constructor(
+        address _core,
+        address[] memory _sources,
+        uint256[] memory _thresholds
+    ) 
+        CoreRef(_core)
+    {
+        require(_sources.length == _thresholds.length, "array length mismatch");
+        require(_sources.length > 0, "must provide at least once source");
+        
+        for(uint i=0; i<_sources.length; i++) {
+            sources.add(_sources[i]);
+            thresholds[_sources[i]] = _thresholds[i];
+            emit SourceAdded(_sources[i], _thresholds[i]);
+        }
+    }
+
+    /// @notice convenience method to view sources & thresholds
+    function getSources() external view returns (address[] memory sourceAddresses, uint256[] memory sourceThresholds) {
```

Comment:
returning an array of structs seems cleaner than returning 2 separate arrays of equal length.